pkg/cvo/internal/operatorstatus.go

```diff
@@ -232,9 +239,10 @@ func waitForOperatorStatusToBeDone(ctx context.Context, interval time.Duration,
 			Nested: fmt.Errorf("cluster operator %s is not done; it is available=%v, progressing=%v, degraded=%v",
 				actual.Name, available, progressing, degraded,
 			),
-			Reason:  "ClusterOperatorNotAvailable",
-			Message: fmt.Sprintf("Cluster operator %s has not yet reported success", actual.Name),
-			Name:    actual.Name,
+			UpgradeSucceeding: true,
```

Comment:
This case is a pretty big bucket, and I'm not sure all of the situations that hit this code are actually succeeding.  Having `Available=False` seems like it deserves `UpgradeSucceeding: false`.  Do we have other cases in here?  `ClusterOperatorNotAvailable` suggests that the original dev was mostly worried about `Available=False`, but we should probably go through the code and see if there are cases like `Available=True` but `Degraded=True` that need a more specific `Reason` and which would qualify for the less-alarming `UpgradeSucceeding: true`.

Reply:
This now results in failing if `Available=False`